src/Http/Client/Response.php

```diff
@@ -453,11 +434,43 @@ public function getCookie($name)
      */
     public function getCookieData($name)
     {
-        if (!isset($this->_cookies[$name])) {
+        $this->buildCookieCollection();
+
+        if (!$this->cookies->has($name)) {
             return null;
         }
 
-        return $this->_cookies[$name];
+        return $this->cookies->get($name)->toArrayCompat();
+    }
+
+    /**
+     * Lazily build the CookieCollection and cookie objects from the response header
+     *
+     * @return void
+     */
+    protected function buildCookieCollection()
+    {
+        if ($this->cookies) {
+            return;
+        }
+        $this->cookies = CookieCollection::createFromHeader($this->getHeader('Set-Cookie'));
+    }
+
+    /**
+     * Property accessor for `$this->cookies`
+     *
+     * @return array Array of Cookie data.
+     */
+    protected function _getCookies()
```

Comment:
This is for the magic properties that the response exposes. `$response->cookies` is provided by this method and the _get` prefix is part of the required convention for exposing properties through methods on this class.